modules/ml/src/main/java/org/apache/ignite/ml/recommendation/RecommendationModel.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.recommendation;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Map;
+import org.apache.ignite.ml.IgniteModel;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+
+/**
+ * Recommendation model that predicts rating for {@link ObjectSubjectPair}.
+ *
+ * @param <O> Type of an object of recommendation.
+ * @param <S> Type of a subject of recommendation.
+ */
+public class RecommendationModel<O extends Serializable, S extends Serializable>
+    implements IgniteModel<ObjectSubjectPair<O, S>, Double> {
+    /** */
+    private static final long serialVersionUID = -3664382168079054785L;
+
+    /** Object of recommendation matrix (result of factorization of rating matrix). */
+    private final Map<O, Vector> objMatrix;
```

Comment:
Please be careful with size of this matrices
I suppose that this model can be very big in terms of memory using. Lets create additional ticket for another implementation where model is distributed on cluster.

Reply:
Correspondent task created: https://ggsystems.atlassian.net/browse/GG-22331.